### ISSUE TITLE
'Share on Codepad' contex menu in the Python Console Editor throws a python stacktrace

### ISSUE THREAD

**Issue by rduivenvoorde** [qgis/QGIS#37574]
While fiddling with https://github.com/qgis/QGIS/pull/37573 I tested the context menu 'Share on Codepad' which throws a stacktrace.

Tested both in Master and 3.14

![Screenshot-20200703202112-531x586](https://user-images.githubusercontent.com/731673/86491314-c35c2380-bd6a-11ea-8f9c-b1c1e4f2954a.png)

This throws:

```
Traceback (most recent call last):
              File "/home/richard/git/build-qgisqt-Desktop-Debug/output/python/console/console_editor.py", line 479, in codepad
              response = urllib.request.urlopen(url, urllib.parse.urlencode(values))
              File "/usr/lib/python3.8/urllib/request.py", line 222, in urlopen
              return opener.open(url, data, timeout)
              File "/usr/lib/python3.8/urllib/request.py", line 522, in open
              req = meth(req)
              File "/usr/lib/python3.8/urllib/request.py", line 1281, in do_request_
              raise TypeError(msg)
             TypeError: POST data should be bytes, an iterable of bytes, or a file object. It cannot be of type str.
```

**Comment by agiudiceandrea** [comment]
I think the error could be avoided changing line 479 in console_editort.py from
```python
response = urllib.request.urlopen(url, urllib.parse.urlencode(values))
```

to
```python
response = urllib.request.urlopen(url, urllib.parse.urlencode(values).encode('utf-8'))
```

Anyway, it seems now codepad.org is protected by reCAPTCHA, so the server would not return the link anyway...

**Comment by nyalldawson** [comment]
Could we swap it out for gist?

**Comment by agiudiceandrea** [comment]
I think the error could be avoided simply changing line 479 in console_editort.py from
```python
response = urllib.request.urlopen(url, urllib.parse.urlencode(values))
```

to
```python
response = urllib.request.urlopen(url, urllib.parse.urlencode(values).encode('ascii'))
```

It seems now codepad.org is protected by reCAPTCHA, so the server would not return the link anyway...

**Comment by rduivenvoorde** [comment]
@agiudiceandrea yes, I already updated the code, but got stuck because codepad.org does return the opening page, which is indeed protected by reCAPTCHA.

@nyalldawson gists can (for what I just googled) only be made when you login OR with a personal token (which I think you do not want to use in QGIS source code).

If there are no easy ways to publicly share code, I would opt for just remove this menu item. Also as it is apparently seldom used, before I tried this by accident, nobody complaint about it...

**Comment by slarosa** [comment]
We are aware this is not a basic user option, I suppose it is used only by advanced users. 
I like the idea of replacing codepad with gist, so to avoid removing it (and if alternative approach does not exists) maybe we can add an option in settings to allow the "advanced" user to set the token for the github platform. And then only if a token is set the entry menu will be enabled.

**Comment by rduivenvoorde** [comment]
I think we are a Desktop GIS and ( although pretty good ;-) ) not a Python IDE.

@slarosa in my opinion this is a lot of hassle (adding gui elements for it, making it work in the auth framework etc etc) for something that is so easy to do by copy/paste in a browser.

I will do a PR. At least for 3.16 we should hide the menu (as currently it is throwing a stacktrace).
But I'm in favour for cleaning it up for now.

**Comment by slarosa** [comment]
@rduivenvoorde PR is ready here https://github.com/qgis/QGIS/pull/38768 :-), I would prefer to keep the feature, as it only needs of a personal token which most users can have.

**Comment by rduivenvoorde** [comment]
Ah duh. sorry for this noise then, thanks for your work there!!

**Comment by slarosa** [comment]
fixed in #38768
